packages/react-data-grid/src/ReactDataGrid.js

```diff
@@ -251,7 +251,11 @@ const ReactDataGrid = createReactClass({
   },
 
   onPressEnter(e: SyntheticKeyboardEvent) {
-    this.setActive(e.key);
+    const didSetActive = this.setActive(e.key);
+
+    if (!didSetActive) {
```

Comment:
If we didn't set active it means the cell was either currently active or the cell is not editable. In both cases we want to navigate.